SimG4Core/Application/src/StackingAction.cc

```diff
@@ -211,6 +212,18 @@ G4ClassificationOfNewTrack StackingAction::ClassifyNewTrack(const G4Track* aTrac
       // potentially good for tracking
       const double ke = aTrack->GetKineticEnergy();
 
+      auto proc = aTrack->GetCreatorProcess();
+      G4int subType = proc->GetProcessSubType();
+      if (nullptr != proc && subType == 16) {
```

Comment:
```suggestion
      G4int subType = [any reasonable initialization]
      auto proc = aTrack->GetCreatorProcess();
      if (nullptr != proc) {
        G4int subType = proc->GetProcessSubType();
        if (subType == 16) {
```
if you need to check `proc` not being a null pointer

Reply:
@perrotta , thanks, the logic is improved now. For secondary tracks the creator process is always existing, so check is removed. The main update in in line 217-219, which recover MC truth handling for G4GammaGeneralProcess.